### ISSUE TITLE
REST API stop working since Umbraco 7.8.*

### ISSUE THREAD

**Issue by IBesedinBellIntegrator** [umbraco/UmbracoRestApi#42]
Hello.

UmbracoCms.RestApi version=0.9.0

Trying to start new empty project on umbraco 7.8.1 and stuck on using REST API. The new token request executed correctly and returns token string. /content and /content/published requests are broken and return HTTP 401.

Ive tried several versions and collect this table:
7.7.7 - works
7.7.11 - works
7.8.0 - doesn't
7.8.1 - doesn't
My OwinStartup.cs is same for all tests, it's here:
```
public override void Configuration(IAppBuilder app)
        {
            app.UseCors(CorsOptions.AllowAll);

            base.Configuration(app);

            app.ConfigureUmbracoRestApi(new UmbracoRestApiOptions()
            {
                CorsPolicy = new CorsPolicy()
                {
                    AllowAnyHeader = false,
                    AllowAnyMethod = false,
                    AllowAnyOrigin = false,
                }
            });

            app.UseUmbracoBackOfficeTokenAuth();
        }
```

All tests have been done locally, so CORS settings doesn't matter.
Also i tested UseUmbracoCookieAuthenticationForRestApi, it works for all versions in browser.

**Comment by ddineen** [comment]
Same here; I upgraded from 7.7.6 to 7.9.2 and my Umbraco extension rest API authentication is no longer working. Using [UmbracoAuthorize] on the controller breaks the call. The call is too a controller that inherits from the UmbracoApiController. I need to keep the routes the same, for any solution I implement. I can get an access token no problem from Umbraco.  Any ideas / questions?

**Comment by zpqrtbnk** [comment]
(troubleshooting)

**Comment by zpqrtbnk** [comment]
Reproduced. 
- With Cms 7.7.13 + RestApi 0.9.0 + above Owin startup, `/umbraco/oauth/token` returns a token, and then `/umbraco/rest/v1/content` returns some content.
- With Cms 7.8.0 + RestApi 0.9.0 + above Owin startup, `/umbraco/oauth/token` returns a token, but then `/umbraco/rest/v1/content` fails with "authorization has been denied".

**Comment by zpqrtbnk** [comment]
Cause has been identified, work-in-progress.

**Comment by zpqrtbnk** [comment]
Root cause is in Umbraco Core. Fix is currently being reviewed.

**Comment by Shazwazza** [comment]
This is fixed in this Core issue: http://issues.umbraco.org/issue/U4-11087 which is due out for 7.9.3

**Comment by ddineen** [comment]
Wow, that is great. Thank you very much!


Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Shannon Deminick <notifications@github.com> Date: 3/12/18  8:28 AM  (GMT-06:00) To: umbraco/UmbracoRestApi <UmbracoRestApi@noreply.github.com> Cc: ddineen <dannytd@gmail.com>, Comment <comment@noreply.github.com> Subject: Re: [umbraco/UmbracoRestApi] REST API stop working since Umbraco 7.8.* (#42) 
Closed #42.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/umbraco/UmbracoRestApi","title":"umbraco/UmbracoRestApi","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/umbraco/UmbracoRestApi"}},"updates":{"snippets":[{"icon":"DESCRIPTION","message":"Closed #42."}],"action":{"name":"View Issue","url":"https://github.com/umbraco/UmbracoRestApi/issues/42#event-1516215301"}}}
